free-programming-books.md

```diff
@@ -457,6 +457,7 @@
 * [Computer Musings](http://scpd.stanford.edu/knuth/index.jsp) (lectures by Donald Knuth)
 * [CS For All](https://www.cs.hmc.edu/csforall/)
 * [Culture \& Empire: Digital Revolution](http://hintjens.com/books) (PDF)
+* [Dependency Injection: A Practical Introduction](http://www.jeremybytes.com/downloads/dependencyinjection.pdf) (PDF)
```

Comment:
One issue we have here is whether something is enough of a book to merit listing; we can't list helpful articles, for example, because there are just too many. On the author's website, he labels this as a "walkthough", and doesn't include it on his "bookshelf". What do you think, is this better described as an "article", or as a "book"?